fastai/vision/data.py

```diff
@@ -58,7 +58,10 @@ def normalize(x:TensorImage, mean:FloatTensor,std:FloatTensor)->TensorImage:
 
 def denormalize(x:TensorImage, mean:FloatTensor,std:FloatTensor, do_x:bool=True)->TensorImage:
     "Denormalize `x` with `mean` and `std`."
-    return x.cpu()*std[...,None,None] + mean[...,None,None] if do_x else x.cpu()
+    if x.dtype == torch.float16:
+        return x.cpu().float()*std[...,None,None] + mean[...,None,None] if do_x else x.cpu()
+    else:
```

Comment:
You don't need to do a test: x.cpu().float() won't do anything if x is already in FP32 precision.

Reply:
Got it. I've changed it as suggested.